javascript_nodejs/6.using-cards/bot.js

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityTypes, CardFactory } = require('botbuilder');
+const { ChoicePrompt, DialogSet, DialogTurnResult, DialogTurnStatus, ListStyle } = require('botbuilder-dialogs');
+
+/**
+ * RichCardsBot prompts a user to select a Rich Card and then returns the card that matches the user's selection.
+ */
+class RichCardsBot {
+    
+    /**
+     * Constructs the three pieces necessary by this bot to operate:
+     * 1. StatePropertyAccessor
+     * 2. DialogSet
+     * 3. ChoicePrompt
+     * 
+     * The only argument taken by this constructor is a ConversationState instance, although any BotState instance would suffice for this bot. `conversationState` is used to create a StatePropertyAccessor, which is needed to create a DialogSet. All botbuilder-dialogs `Prompts` need a DialogSet to operate.
```

Comment:
>e only argument taken by this constructor is a ConversationState instance, although any BotState instance would suffice for this bot [](start = 9, length = 132)

This is confusing. "We did X, but Y would have been enough, but we like to confuse you so we really thought about Z". Simplify. :)

Reply:
Changed, but feedback still welcome...